mantidproject/mantid
# Load event nexus from ILL

**Description of work.**


Make the necessary changes so that ILL's event data can be loaded as event nexuses by Mantid.
Also, in order to start allowing for asked data treatment of it, add an option in GenerateEventsFilter for logarithmic and reverse logarithmic time split, and pass it to the FilterEvents interface. 


**To test:**
- Load `UnitTest/ILL/D22B/042730_trunc.event.nxs`, checking `LoadMonitors`.
- Check there are 1000 events in the workspace.
- Check that the monitor events have been loaded.
- Open the instrument viewer, check the integration boundaries are between 23.71 and 59117.1, and that the instrument looks reasonable.
- Open the FilterEvents interface. Select the workspace, and in the `FilterByTime` tab, check there is a `UseReverseLogarithmic` tick box. Filter by some time interval, check the result is correct. Filter by logarithmic and reverse logarithmic, check the results look sensible (note that reverse logarithmic may not work exactly as expected, due to the way time is handled in `GenerateEventsFilter`. The point is that the bins should expand exponentially starting from the end, not the exact values of this expansion). Check the duration of the created workspaces have been updated to reflect the split.

- Reload the same file after checking `LoadAllLogs`. Check that the sample logs are now filled with a lot of logs. Check that the `start_time` and `end_time` have been correctly loaded. 

- Check quickly that those same steps also work for `ILL/D22B/000242_trunc.event.nxs`.


WIP for #31747 .






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
